source/slang/slang-lower-to-ir.cpp

```diff
@@ -776,6 +778,24 @@ LoweredValInfo emitCallToDeclRef(
         }
         else
         {
+            switch (intrinsicOp)
+            {
+                case kIROp_GetStringHash:
+                {
+                    IRStringLit* stringLit = as<IRStringLit>(args[0]);
+
+                    if (stringLit == nullptr || stringLit->getStringSlice() == UnownedStringSlice())
```

Comment:
The second tests seems to be ruling out calling `getStringHash("")` with an empty string. Is there a reason for that to be illegal?

Reply:
It's a side effect of using StringSlicePool. That nullptr and "" are special cased and assigned at construction.

Reply:
So that makes two implementation details of `StringSlicePool` that leak into the public API:

1. The strings returned via reflection aren't guaranteed null-terminated, even though that would be the more humane design choice.

2. The `getStringHash` function in Slang can't be called with an empty string literal, with no clear reason why from the user's perspective (after all, an empty string can certainly be hashed...).

A simpler (but slower) implementation using a `List<String>` wouldn't have either of those limitations, and I'd consider that the baseline that a more optimized implementation needs to improve on.

Reply:
Both are resolved in a review PR.